### PR TITLE
feat(vue): web-types support

### LINKED ISSUE #19522: Generate web-types for @ionic/vue package
# Feature Request

**Ionic version:**

[x] @ionic/vue 0.0.x

**Feature Request**

This is a feature request to include [`web-types`](https://github.com/JetBrains/web-types) metadata information with `@ionic/vue` library to provide users of WebStorm (and possibly other IDEs in the future) with precise code completion, inspections and documentation for Ionic components.

**Preferred Solution**

The best solution would be to generate the file based on existing Ionic documentation JSONs during the build process and ship it with the library. This solution has been implemented in `quasar` library and are being worked on in `vuetify` and `bootstrap-vue`. I've looked at some of the JSONs generated by ionic documentation and it should be relatively straightforward to transform information to a slightly different format. 

I can work on the transformation code itself, but it would be pretty difficult for me to hook into the build process in the correct place, so I need some help from your side, before I can create a PR for this. 

**Alternatives**

It's difficult for an IDE to analyze contents of compiled Vue library and the quality of code completion, inspection and documentation is pretty poor. `web-types` offer excellent opportunity to provide consumers of your library with precise information. It is especially useful for new users of the library as it allows to avoid constant looking into documentation on the web.

**Related Code**

Here is an example PR from Quasar - https://github.com/quasarframework/quasar/pull/4749

**Additional Context**

The feature has been requested by one of our users on Twitter: https://twitter.com/leereichardt/status/1165554297225891841

### PR DESCRIPTION
**Pull request by piotrtomiak**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: resolves #19522 

## What is the new behavior?

This PR introduces bundling of web-types with `@ionic/vue` for in-editor documentation support. The web-types standard is an open source standard supported so far only by WebStorm and is bundled with various other Vue libraries, like `vuetify`, `bootstrap-vue` or `quasar`. 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by liamdebeasi** [comment]
Thank you! This looks great. I created a dev build and tested it in the latest version of WebStorm, and it is working well:

```
npm install @ionic/vue@5.5.0-dev.202011041734.2bd24d6 @ionic/vue-router@5.5.0-dev.202011041734.2bd24d6 --save-exact
```

File: packages/vue/scripts/build-web-types.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+const fs = require("fs")
+
+// Web-types build require docs to be built first
+const docs = require("../../../docs/core.json")
```
Comment (liamdebeasi) [inline packages/vue/scripts/build-web-types.js:4]:
```suggestion
const docs = require('@ionic/core/dist/docs.json')
```

Let's use the docs.json file output by ionic core as that is what we use in the other copy scripts (same exact file, just loading from a different location).

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
One small change, otherwise looks great.

**Review by liamdebeasi — APPROVED** [review]
Thank you for this! I will make sure this gets into our next minor release.

**Comment by liamdebeasi** [comment]
Merged. Thank you! This will be in our next minor release.

**Comment by piotrtomiak** [comment]
Thank you!

**Comment by aparajita** [comment]
Thank you @piotrtomiak!

**Comment by ddesmarais749** [comment]
I know this commit is old but seems like it is still an issues with Ionic v6.1.5. I can't get Webstorm autocomplete work.

Web-types.json is empty

The web-type.json in the "@ionic" node_modules looks like this :

image

Can anyone help me?

**Pull request merged**